arbnode/sequencer.go

```diff
@@ -114,15 +122,94 @@ func (i *txQueueItem) returnResult(err error) {
 	close(i.resultChan)
 }
 
+type nonceCache struct {
+	cache *containers.LruCache[common.Address, uint64]
+	block common.Hash
+	dirty *types.Header
+}
+
+func newNonceCache(size int) *nonceCache {
+	return &nonceCache{
+		cache: containers.NewLruCache[common.Address, uint64](size),
+		block: common.Hash{},
+		dirty: nil,
+	}
+}
+
+func (c *nonceCache) Matches(header *types.Header) bool {
```

Comment:
Matches is only called from inside nonceCache.. pointer comparison is already a little weird, might make sense to make Matches private.